File: src/striver/binary_search/p704/p704.go
package main

func binarySearch(nums []int, low, high, target int) int {
	if low > high {
		return -1
	}

	mid := (low + high) / 2
	if nums[mid] == target {
		return mid
	} else if target > nums[mid] {
		return binarySearch(nums, mid+1, high, target)
	} else {
		return binarySearch(nums, low, mid-1, target)
	}

}

func search(nums []int, target int) int {
	return binarySearch(nums, 0, len(nums)-1, target)
}

func main() {

}

File: striver/binary_search/p704.py
# https://leetcode.com/problems/binary-search/description/

from typing import List 

class Solution:
    def search(self, arr: List[int], target: int) -> int:
        n = len(arr)
        low = 0 
        high = n - 1

        while low <= high:
            mid = (low+high)//2
            if arr[mid] == target:
                return mid 
            elif arr[mid] < target:
                low = mid + 1
            else:
                high = mid - 1
        
        return -1
        
